datafusion/core/src/optimizer/utils.rs

```diff
@@ -223,6 +225,17 @@ pub fn from_plan(
             let right = &inputs[1];
             LogicalPlanBuilder::from(left).cross_join(right)?.build()
         }
+        LogicalPlan::Subquery(Subquery { .. }) => {
+            // let schema = inputs[0].schema().as_ref().clone().into();
+            // let schema =
+            //     DFSchemaRef::new(DFSchema::try_from_qualified_schema(alias, &schema)?);
+            // Ok(LogicalPlan::Subquery(Subquery {
+            //     subquery
+            //     input: Arc::new(inputs[0].clone()),
+            //     schema,
+            // }))
+            Err(DataFusionError::Plan("not implemented".to_string()))
```

Comment:
```suggestion
            Err(DataFusionError::Plan("subquery support is not full implemented yet".to_string()))
```

Reply:
I have now implemented this section of code, although it is not exercised by any tests yet